part2/src/index.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+
+
+
+const Header = ({ course }) => (
+  <h1> {course} </h1>
+)
+
+
+const Part = ({ name, exercises }) => (
+  <p>
+    {name} {exercises}
+  </p>
+)
+
+const Total = ({ parts }) => { 
+
+  const total = parts.reduce((tot, arr) =>
+     tot + arr.exercises, 0)
+
+  return (
+<>
+  <h4> Number of exercises: {total}</h4>
+</>
+  )
+}
+
+
+
+const Content = ({ parts }) => {
+
+  return (
+    <>
+      {parts.map((part) => (
+        <Part name={part.name} exercises={part.exercises} key={part.id} />
+      ))}
+    </>
```

Comment:
Šeit var ar fragmentu, kas nav slikti, jo labi nes meaningu, bet [React (JSX) pieļauj arī atgriezt komponenšu masīvu](https://reactjs.org/docs/jsx-in-depth.html#jsx-children:~:text=A%20React%20component%20can%20also%20return%20an%20array%20of%20elements%3A), līdz ar to varētu arī maukt: `return parts.map(part => <Part /**/ />)` ;)

Just saying, manuprāt no koda sapratnes viedokļa šādā gadījumā ar fragmentu ir labāk.

P.S. Paskaties kopumā to linku, tur visādi interesanti patterni.